### PR TITLE
Fix undefined behavior in ZSTD_decompressStream()

### PR DESCRIPTION
**Pull request by daniellerozenblit**

### CODE REVIEW

**Comment by daniellerozenblit** [comment]
## TLDR
This PR fixes two instances of undefined behavior (0 offset to Null pointer) in ZSTD_decompressStream(). Also it adds some additional clang ubsan CI jobs, in addition to existing gcc ubsan jobs.

## Reproduce Bug
Run `CC=clang make uasan-test-zstream` with clang11 or higher. Make sure to add the test case that was added in this PR (in tests/zstreamtest.c) since it exposes one of these errors.

## Benchmarking
This bug fix does not seem to negatively impact decompression performance.

I benchmarked on an Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz machine with core isolation and turbo disabled. I measured decompression time for silesia.tar (203M), compiling with GCC11 and clang15. I ran each scenario 5 times interleaved to eliminate bias from background processes. 

### Clang Results
original: 0m0.472s = 430M/s
this PR: 0m0.469s = 433M/s

This PR increased compression speed by around 0.698%, probably due to random noise.

### GCC Results
original: 0m0.466s = 436M/s
this PR: 0m0.466s = 436M/s

This PR did not change compression speed (0.00% decrease).
